### PR TITLE
Enable runtime and sphinx-level signatures for ops API

### PR DESCRIPTION
**Pull request by klecki**
## Description:
This should allow proper rendering of the parameter references in the ops API part of the documentation. Additionally this fixes lack of default values for some CMAKE build flags.

If the `BUILD_WITH_ASAN` doesn't have a value
`if(NOT ${CMAKE_CROSSCOMPILING} AND NOT ${BUILD_WITH_ASAN})`  does not evaluate to true and `dali/python/CMakeLists.txt` skips the generation of interface files with API signatures - breaking local development.

See:
https://github.com/NVIDIA/DALI/blob/main/dali/python/CMakeLists.txt#L62

### Affected modules and functionalities:
Signatures in ops API + CMake.

### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [x] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: dali/python/nvidia/dali/ops/_docs.py [hunk 1]
```diff
@@ -256,14 +254,12 @@ def _docstring_prefix_auto(op_name):
...
-        return """__call__(**kwargs)
-
+        return """
 Operator call to be used in graph definition. This operator doesn't have any inputs.
 """
     elif schema.MaxNumInput() == 1:
         input_name = _names._get_input_name(schema, 0)
-        ret = f"""__call__({input_name}, **kwargs)
-
+        ret = f"""
```
Comment (klecki) [inline dali/python/nvidia/dali/ops/_docs.py:262]:
```suggestion
        ret = """
```

**Review by jantonguirao — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -43,6 +43,10 @@ option(WITH_DYNAMIC_CUDA_TOOLKIT "Links CUDA toolkit libraries dynamically (NPP,
 
 option(WITH_DYNAMIC_NVIMGCODEC "Links nvimgcodec library dynamically during runtime" ON)
 
+option(BUILD_WITH_ASAN "Build with ASAN" OFF)
+option(BUILD_WITH_LSAN "Build with LSAN" OFF)
+option(BUILD_WITH_UBSAN "Build with UBSAN" OFF)
+
```
Comment (mzient) [inline CMakeLists.txt:49]: Is it related?

File: dali/python/nvidia/dali/ops/__init__.py [hunk 3]
```diff
@@ -681,6 +681,25 @@ def _operator_name(self):
...
+            data_node_return=False,
+            all_args_optional=True,
+        )
+        Operator.__call__.__signature__ = _signatures._call_signature(
+            schema,
+            include_inputs=True,
+            include_kwargs=True,
+            include_self=True,
+            all_args_optional=True,
+        )
```
Comment (mzient) [inline dali/python/nvidia/dali/ops/__init__.py:702]: Now if we replace *inputs with named arguments - are they marked as positional-only?

File: CMakeLists.txt (hunk 2 above)
Comment (klecki) [reply CMakeLists.txt:49]: Kinda, without it the signatures are not generated in a default build

File: dali/python/nvidia/dali/ops/__init__.py (hunk 3 above)
Comment (klecki) [reply dali/python/nvidia/dali/ops/__init__.py:702]: This is how they are already defined in the pyi files, and yes, all inputs are positional only

**Review by mzient — APPROVED** [review]

**Pull request merged**